lib/toolkit/loader.js

```diff
@@ -785,7 +786,20 @@ function Loader(options) {
     if (isNative && !uri)
       uri = id;
     let module = Module(id, uri);
-    module.exports = freeze(modules[id]);
+
+    // Lazily expose built-in modules in order to
+    // allow them to be loaded lazily.
+    let cachedModules = modules;
+    Object.defineProperty(module, "exports", {
+      enumerable: true,
+      get: function() {
+        return cachedModules[id];
+      },
+      set: function(v) {
```

Comment:
Is there a use case for the setter ? If so please add a comment explaining it. Although I really hope we don't need a setter here, in which case I'd suggest to either remove it or make one that throws exception explaining why this is forbidden.